homeassistant/components/binary_sensor/__init__.py

```diff
@@ -93,6 +93,7 @@
 
 # On means open, Off means closed
 DEVICE_CLASS_WINDOW = "window"
+DEVICE_CLASS_WINDOW_VARIANT = "window_variant"
```

Comment:
This is a change the entity model, which needs to be discussed in https://github.com/home-assistant/architecture first.

Have a look at: http://developers.home-assistant.io/docs/core/entity/#changing-the-entity-model. 


>If you want to add a new feature to an entity or any of its subtypes (light, switch, etc), you will need to propose it first in our architecture repo. Only additions will be considered that are common features among various vendors.

Reply:
As @iMicknl mentioned above, this needs an architectural change to move forward.

Reply:
Done.

I though that the entity model was more based on the `list` of properties than on the `content` of each property.

Simone